spacebench/algorithms/dapsm.py

```diff
@@ -0,0 +1,115 @@
+import pandas as pd
+import numpy as np
+import sys
+from scipy.optimize import linear_sum_assignment
+
+def StandDist(x):
+    standx = (x - np.min(x))/(np.max(x) - np.min(x))
+    return(standx)    
+
+
+# Modify dist_ps so it just takes in a distance matrix
+def dist_ps(dist_mat, ps_diff, caliper, caliper_type, distance = StandDist,
+            weight=0.8,
+            matching_algorithm='optimal'):
+    # This line is not needed if distances are already normalized between 0 and 1
+    stand_dist_mat = distance(dist_mat)
+
+    # DAPS matrix
+    dapscore = (1-weight)*stand_dist_mat + weight*ps_diff
+
+    # Creating the matrix we will use for matching, depending on
+    # whether the caliper is set on DAPS or on the PS.
+    if caliper_type == 'DAPS':
+        dapscore[dapscore > caliper * np.std(dapscore)] = sys.float_info.max
+    elif caliper_type == 'PS':
+        dapscore[ps_diff > caliper * np.std(ps_diff)] = sys.float_info.max
+
+    if matching_algorithm == 'greedy':
+        # TO DO
```

Comment:
Combine "TODO's" into one word, so that numerous document editing tools can easily detect them later.

```
#TO DO --> #TODO
```